### ISSUE TITLE
Can't build CarthageKit.framework with carthage build

### ISSUE THREAD

**Issue by dcaunt** [Carthage/Carthage#658]
I am trying to build `CarthageKit.framework` from the command line with `carthage build` but it is failing due to a missing module in a dependency of CarthageKit. I also tried to build Commandant and some of its dependencies in a similar manner and these fail too. What am I doing wrong?

I can build CarthageKit in Xcode from the workspace just fine, and `xcodebuild -scheme CarthageKit -workspace Carthage.xcworkspace build` works too.

I'm running the following commands to build:

```
git clone git@github.com:Carthage/Carthage.git
cd Carthage
script/bootstrap
carthage build --no-skip-current
```

Carthage fails to build with the following output:

```
$ carthage build --no-skip-current
*** xcodebuild output can be found in /var/folders/t6/566y64zd1c54v7tkfq9jx4300000gn/T/carthage-xcodebuild.mmfRbw.log
*** Building scheme "Box-iOS" in Box.xcodeproj
*** Building scheme "Box-Mac" in Box.xcodeproj
*** Building scheme "Nimble-iOS" in Nimble.xcodeproj
*** Building scheme "Nimble-OSX" in Nimble.xcodeproj
*** Building scheme "PrettyColors iOS" in PrettyColors.xcodeproj
*** Building scheme "PrettyColors Mac" in PrettyColors.xcodeproj
*** Building scheme "Quick-iOS" in Quick.xcworkspace
*** Building scheme "Quick-OSX" in Quick.xcworkspace
*** Building scheme "Runes-iOS" in Runes.xcodeproj
*** Building scheme "Runes-Mac" in Runes.xcodeproj
*** Building scheme "ReactiveTask" in ReactiveTask.xcworkspace
*** Building scheme "Result-iOS" in Result.xcworkspace
*** Building scheme "Result-Mac" in Result.xcworkspace
*** Building scheme "Commandant" in Commandant.xcworkspace
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Option.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Errors.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Switch.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Command.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/ArgumentParser.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(7 failures)
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest, missing required architecture armv7 in file /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest (2 slices)
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: ignoring file /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest, missing required architecture arm64 in file /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest (2 slices)
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
ld: warning: embedded dylibs/frameworks only run on iOS 8 or later
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
/Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift:10:8: error: no such module 'Result'
A shell task failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/HelpCommand.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Option.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Errors.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Switch.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/Command.swift
	CompileSwift normal x86_64 /Users/dcaunt/Projects/Carthage/Carthage/Checkouts/Commandant/Commandant/ArgumentParser.swift
	CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
(7 failures)
```

I have tried a few different commands for bringing in the submodules – the reason for trying to build CarthageKit in the first place was to better understand how Carthage can manage submodules.

Thanks

**Comment by mdiep** [comment]
I'm not sure off the top of my head. We don't use `carthage` to build `Carthage` at the moment (although we do use it to manage dependencies).

**Comment by ikesyo** [comment]
This is fixed in Carthage/Commandant#27. We should cut a new Commandant release, and use it in Carthage.

**Comment by ikesyo** [comment]
I've just released [Commandant 0.6.1](https://github.com/Carthage/Commandant/releases/tag/0.6.1).

**Comment by dcaunt** [comment]
Great stuff, CarthageKit builds! Thanks @ikesyo
